### PR TITLE
Add parser function to create json_object and rename json_object to create a standard.

### LINKED ISSUE #43: Create parser function to cloud_unregister_device
A function `parser_unregister_json_create` should be called to avoid cloud source file call JSON-c methods.

### PR DESCRIPTION
**Pull request by vitbaq**
Add parser function to create json_object on cloud_unregister_device.
Rename json_object on cloud_auth_device to create a standard.
Closes #43

### CODE REVIEW

File: src/cloud.c [hunk 1]
```diff
@@ -311,13 +311,12 @@ int cloud_register_device(const char *id, const char *name)
...
 	amqp_bytes_t queue_cloud;
-	json_object *jobj;
+	json_object *jobj_unreg;
 	const char *json_str;
 	int result;
 
-	jobj = json_object_new_object();
-	json_object_object_add(jobj, "id", json_object_new_string(id));
-	json_str = json_object_to_json_string(jobj);
+	jobj_unreg = parser_unregister_json_create(id);
```
Comment (ramonhpr) [inline src/cloud.c]: Verify if `jobj_unreg` is NULL

File: src/cloud.c [hunk 2]
```diff
@@ -349,7 +349,7 @@ int cloud_unregister_device(const char *id)
 int cloud_auth_device(const char *id, const char *token)
 {
 	amqp_bytes_t queue_cloud;
-	json_object *jobj_device;
```
Comment (ramonhpr) [inline src/cloud.c]: About commit message:
This patch changes **json_object** name on cloud_auth_device **to** a more
suitable one. This **json_object** **represents** **an** auth message.

File: src/cloud.c [hunk 3]
```diff
@@ -359,8 +359,8 @@ int cloud_auth_device(const char *id, const char *token)
 		return -1;
 	}
 
-	jobj_device = parser_auth_json_create(id, token);
-	json_str = json_object_to_json_string(jobj_device);
+	jobj_auth = parser_auth_json_create(id, token);
```
Comment (ramonhpr) [inline src/cloud.c]: verify if `jobj_auth` is NULL

Comment (ramonhpr) [reply src/cloud.c]: In another commit

File: src/cloud.c (hunk 1 above)
Comment (vitbaq) [reply src/cloud.c]: Ok

File: src/cloud.c (hunk 2 above)
Comment (vitbaq) [reply src/cloud.c]: Ok

File: src/cloud.c (hunk 3 above)
Comment (vitbaq) [reply src/cloud.c]: Ok

File: src/cloud.c [hunk 4]
```diff
@@ -283,6 +283,10 @@ int cloud_register_device(const char *id, const char *name)
 	}
 
 	jobj_device = parser_device_json_create(id, name);
+	if (!jobj_device) {
+		amqp_bytes_free(queue_cloud);
+		return -1;
```
Comment (ramonhpr) [inline src/cloud.c]: Since this code will be sent to things. We should use one of the knot code err.
As an example `KNOT_ERR_CLOUD_FAILURE`

Comment (vitbaq) [reply src/cloud.c]: Ok

**Pull request merged**